src/CommunityToolkit.Maui.Core/Layouts/DockLayoutManager.shared.cs

```diff
@@ -81,21 +81,19 @@ public override Size ArrangeChildren(Rect bounds)
 		var width = bounds.Width - padding.HorizontalThickness;
 		var height = bounds.Height - padding.VerticalThickness;
 
-		var i = 0;
-		foreach (var child in dockLayout)
+		for (var i = 0; i < dockLayout.Count; i++)
 		{
+			var child = dockLayout[i];
 			if (child.Visibility != Visibility.Visible)
 			{
 				continue;
 			}
 
-			i++;
-
 			var childWidth = Math.Min(width, child.DesiredSize.Width);
 			var childHeight = Math.Min(height, child.DesiredSize.Height);
 
-			var lastItem = (i == dockLayout.Count);
-			if (lastItem && dockLayout.LastChildFill)
+			var isLastChild = (i == dockLayout.Count - 1);
```

Comment:
Does this syntax work? This is just for readability

```suggestion
			var isLastChild = (i == dockLayout[^1]);
```

Reply:
I guess you mean `var isLastChild = (child == dockLayout[^1]);`?

Reply:
Then, I can do `foreach` again (as the index is no longer needed).

Reply:
Nevermind, I just read this wrong.